pgjdbc/src/main/java/org/postgresql/util/GT.java

```diff
@@ -28,7 +29,7 @@ public static String tr(String message, Object... args) {
 
   private GT() {
     try {
-      _bundle = ResourceBundle.getBundle("org.postgresql.translation.messages");
+      _bundle = ResourceBundle.getBundle("org.postgresql.translation.messages", Locale.getDefault(Locale.Category.DISPLAY));
```

Comment:
You could use conditional comment to make extra argument invisible to Java 1.6 like in https://github.com/pgjdbc/pgjdbc/blob/8be516d47ece60b7aeba5a9474b5cac1d538a04a/ubenchmark/src/main/java/org/postgresql/benchmark/statement/FinalizeStatement.java#L64-L66

Reply:
Executing Locale.setDefault(Locale.XXX) also changes the value for Locale.getDefault(Locale.Category.DISPLAY).  So with the proposed change the default behaviour remains the same: those who set default locale will get the same value for both Locale.getDefault() and Locale.getDefault(Locale.Category.DISPLAY).

But it adds possibility for interested parties to change Locale.Category.DISPLAY to another value and affect only messages language.

Thanks for pointing out a trick with maven's #if.
I'll adopt that suggestion.

What kind of test do you want?  The behaviour remains the same with the exception that it will be possible to change locale used only for displaying messages and have another value for collation, number formatting, and so on.